api/admin/controller/collection_settings.py

```diff
@@ -174,7 +174,9 @@ def validate_form_fields(self, is_new, protocols, **fields):
             return MISSING_COLLECTION_NAME
         if "collection" in fields:
             if fields.get("collection"):
-                return self.validate_collection(**fields)
+                invalid_collection = self.validate_collection(**fields)
+                if invalid_collection:
+                    return invalid_collection
```

Comment:
The other changes here are tested by your changes to test_collections, but this feels like it needs an extra test. Is that right?

Like, you would test the case where the collection information is valid but the protocol is missing. Previously it would have passed validation but with this change validation will fail.

Reply:
It looks like those cases are already being tested, at the beginning of `test_collections_post_errors` (line 190 et seq.)...no?  The bug turned out to be caused by `validate_collection` returning even when there wasn't an error; the rest of the validation methods, including the numeric one, were never getting a chance to run.  So, the only resulting change in functionality is that, previously, collections with invalid information would have passed validation--because the code would always have returned after checking for the protocol, and not gone on to check the information--and now they don't; a collection with no protocol wouldn't ever have passed validation.

Reply:
That makes sense.